### PR TITLE
Fix: sibling animations triggered by unmount

### PR DESCRIPTION
**Pull request by shuangq**
Previously we fixed the sibling animation triggered by an unmounting element by calling `didUpdate` in a React effect cleanup, but that no longer works because `checkUpdateFailed` cancel the update before the effect runs.
This PR moves it into `scheduleCheckUpdateFailed` (renamed to `scheduleCheckAfterUnmount`) so that we either run `didUpdate` or clear all the snapshots.

### CODE REVIEW

File: src/projection/node/create-projection-node.ts [hunk 1]
```diff
@@ -566,8 +564,19 @@ export function createProjectionNode<I>({
...
-        scheduleUpdateFailedCheck() {
-            sync.postRender(this.checkUpdateFailed)
+        scheduleCheckAfterUnmount(nodeInLayoutGroup?: IProjectionNode) {
+            /**
+             * If the unmounting node is in a layoutGroup and did trigger a willUpdate,
+             * we manually call didUpdate to give a chance to the siblings to animate.
+             * Otherwise, cleanup all snapshots to prevents future nodes from reusing them.
+             */
+            sync.postRender(() => {
+                if (nodeInLayoutGroup?.isLayoutDirty) {
```
Comment (mattgperry) [inline src/projection/node/create-projection-node.ts:574]: Currently this fires on `root` but that was only so multiple elements could schedule just one callback that removed all snapshots.

If you're doing it this way I think you can just call `projection.schedueCheckAfterUnmount`

File: src/projection/node/create-projection-node.ts [hunk 2]
```diff
@@ -399,9 +399,8 @@ export function createProjectionNode<I>({
                         hasLayoutChanged,
                         layout: newLayout,
                     }: LayoutUpdateData) => {
-                        const dragControls = elementDragControls.get(
-                            visualElement
-                        )
+                        const dragControls =
+                            elementDragControls.get(visualElement)
```
Comment (shuangq) [inline src/projection/node/create-projection-node.ts:403]: @mattgperry hmm now the precommit hook starts to introduce these formatting changes, do you have any idea why?

**Review by mattgperry — APPROVED** [review]

File: src/projection/node/create-projection-node.ts (hunk 2 above)
Comment (mattgperry) [reply src/projection/node/create-projection-node.ts:403]: No its weird I wonder if Prettier has had a quiet update

**Comment by mattgperry** [comment]
Does the full test suite pass locally?

File: src/projection/node/create-projection-node.ts (hunk 1 above)
Comment (shuangq) [reply src/projection/node/create-projection-node.ts:574]: I didn't quite understand, don't we want to same for `didUpdate` if multiple elements schedule it?

**Comment by shuangq** [comment]
> Does the full test suite pass locally?

Yeah it does

File: src/projection/node/create-projection-node.ts (hunk 1 above)
Comment (mattgperry) [reply src/projection/node/create-projection-node.ts:574]: That's a good point but in this case didUpdate could be fired by an unlimited number of elements

**Pull request merged**